Code/Framework/AzCore/Tests/Math/SfmtTests.cpp

```diff
@@ -107,6 +109,82 @@ namespace UnitTest
                 EXPECT_EQ(array64_2[i], g.Rand64());
             }
         }
+
+        template<size_t NumThreads, size_t NumResultsPerThread, typename ResultType>
+        bool ParallelTest()
+        {
+            Sfmt sfmt;
+
+            // Arbitrary but deterministic seed values to guarantee that we get the same random numbers produced every time.
+            AZ::u32 buffer[32] = { 0x8236ed73, 0x854aa369, 0xc7b68864, 0x5f1e49da, 0x58ea5a08, 0xa33fc74b, 0x0336bd81, 0x8d3c11ac,
+                                   0x7312bc57, 0x92fee3d1, 0x4b852f9f, 0xa7ac02ad, 0x4d72fb7a, 0x630641c6, 0x1edbeebf, 0xc18fdabe,
+                                   0xc6588dba, 0x6a821cf5, 0xec7e24b3, 0x44246d7e, 0x24af2f32, 0x4f64d44c, 0x44b24116, 0x65585572,
+                                   0x0f95038d, 0xd3e4c521, 0x6eea6bc1, 0x6d651ab5, 0x25dfc39e, 0x7502d183, 0xd32fc9bf, 0x9854093f };
+
+            // Seed the sfmt generator.
+            sfmt.Seed(buffer, AZ_ARRAY_SIZE(buffer));
+
+            // Create a single array that will hold all the results, but we'll partition it so that each thread will write
+            // its results to a different subset.
+            AZStd::array<ResultType, NumResultsPerThread * NumThreads> results;
+            AZStd::thread threads[NumThreads];
+
+            // Use a semaphore to block thread execution until all the threads are created so that we can synchronize
+            // their start times to the best of our ability.
+            AZStd::semaphore startSync;
+
+            // Spawn a set of threads that will each loop through and call the RandMethod (Rand32 or Rand64) for
+            // a given number of times.
+            for (size_t threadIdx = 0; threadIdx < AZ_ARRAY_SIZE(threads); ++threadIdx)
+            {
+                const size_t startOffset = NumResultsPerThread * threadIdx;
+
+                auto threadFunc = [&startSync, startOffset, &results, &sfmt]()
+                {
+                    // Block so that we can start all the threads at the same time after creation.
+                    startSync.acquire();
+
+                    for (int i = 0; i < NumResultsPerThread; ++i)
+                    {
+                        // Call Rand32 when uint32_t is the passed-in type, and Rand64 if uint64_t is the passed-in type.
+                        if constexpr (AZStd::is_same<ResultType, uint32_t>::value)
```

Comment:
The `_v` shortcut can be used for [AZStd::is_same](https://en.cppreference.com/w/cpp/types/is_same) to save a little bit of typing.
```suggestion
                        if constexpr (AZStd::is_same_v<ResultType, uint32_t>)
```